plugin/pkg/shoot/validator/admission.go

```diff
@@ -354,6 +354,26 @@ func (c *validationContext) validateProjectMembership(a admission.Attributes) er
 	return nil
 }
 
+// validateSeedSelectionForHAShoot validates the seed assigned for the shoot based on the following criteria:
+// 1. A HA shoot with failure tolerance of 'zone' can only be scheduled on multi-zonal seeds
+// 2. A HA shoot with failure tolerance of 'node' can technically be scheduled onto a multi-zonal as well as single-zonal seed. However, we restrict its scheduling only to multi-zonal seeds as of now.
```

Comment:
Maybe I'm missing something, but where is the failure tolerance type validated according to this comment in the below function? I only see handling for the annotation.

Reply:
```go
requestedHAShootControlPlane := shootHasHAAnnotation || c.shoot.Spec.ControlPlane != nil
```
There is currently no check for a specific failure tolerance type. However this will undergo change. Will respond with details as docs

Reply:
Please check again.